### DISCUSSION TITLE
Human-GEM/model/  Human-GEM/code/io/

### DISCUSSION THREAD

**Discussion by helenaherand** [SysBioChalmers/Human-GEM#384] [General]
Are the files in `Human-GEM/model/ `generated from the xml version of the Human1 model? How can I generate the files myself? I looked in the ` Human-GEM/code/io/` folder but could not found the answer to the question.

**Comment by mihai-sysbio** [comment]
That is a very good question @helenaherand, and I hope you don't mind me converting it to a _discussion_.

To put it shortly, the `xml` format gets _created from_ `yml`.

The `develop` branch uses only the `yml` format, and all other file formats are only created when using the `main` branch. The code used to do this is here: https://github.com/SysBioChalmers/Human-GEM/blob/ff109f2283e06bc741ccd991c87ecc11a87a17a6/code/io/increaseHumanGEMVersion.m#L53-L61 

There are, of course, dedicated functions in [`exportHumanGem.m`](https://github.com/SysBioChalmers/Human-GEM/blob/develop/code/io/exportHumanGEM.m), some of which rely on the underlying functions in the RAVEN toolbox.

For more background on how branches are used, see [this discussion](https://github.com/SysBioChalmers/Human-GEM/discussions/377) and also the [Contributing guidelines](https://github.com/SysBioChalmers/Human-GEM/blob/main/.github/CONTRIBUTING.md#branching-model).

Is there something you were wondering about regarding the content of the different file formats?
